### PR TITLE
feat: allow private networks

### PR DESCRIPTION
**Pull request by luislhl**
### Acceptance Criteria
- [ ] It should be possible to use this version of the lib in our Explorer and make it work with private-networks
- [ ] Allow dynamically adding more network configuration options

### Notes
- We are using the same version bytes as the Testnet for the private networks, because we would have problems with TxMiningService if using different ones, since it validates the addresses.

### TODO
- [ ] Release new version of the Explorer once a new version of this lib is available, which should close https://github.com/HathorNetwork/hathor-explorer/issues/125
- [ ] Also release a new version of the Wallet Headless using the new lib version

### CODE REVIEW

File: src/models/network.ts [hunk 1]
```diff
@@ -114,19 +171,50 @@ class Network {
     this.bitcoreNetwork = networkOptions[name];
   }
 
+  /**
+   * Registers a new network configuration.
+   *
+   * We validate the configuration before registering.
+   */
+  static registerNetwork(networkConfig: networkConfigType) {
```
Comment (luislhl) [inline src/models/network.ts:179]: Move this to a different PR, because we should test it further

**Review by msbrogli — APPROVED** [review]
